### PR TITLE
Add a test to load trained model and run prediction for a single sample

### PR DESCRIPTION
**Pull request by pzhanggit**

### CODE REVIEW

File: hydragnn/models/Base.py [hunk 1]
```diff
@@ -244,7 +244,10 @@ def forward(self, data):
 
         #### multi-head decoder part####
         # shared dense layers for graph level output
-        x_graph = global_mean_pool(x, batch.to(x.device))
+        if batch is None:
```
Comment (allaffa) [inline hydragnn/models/Base.py:247]: Nice! I remember having similar issues for other codes every time I called **global_mean_pool** on a single data sample.

File: tests/test_graph_pred.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    log_name = hydragnn.utils.config_utils.get_log_name_config(config)
+    hydragnn.utils.model.load_existing_model(model, log_name)
+
+    isample = random.randrange(len(test_loader.dataset))
+    graph_sample = test_loader.dataset[isample]
+    true_values = graph_sample.y.squeeze()
+    predicted_values = model(graph_sample)
+    yloc = graph_sample.y_loc.squeeze()
+
+    thresholds = 0.75
```
Comment (allaffa) [inline tests/test_graph_pred.py:45]: This threshold every high because the goal of this test is NOT to test the accuracy, but just to make sure that a model is correctly loaded and deployed for predictions. Correct?

**Review by allaffa — COMMENTED** [review]
Just questions to clarify the set up of the unit test.

File: tests/test_graph_pred.py (hunk 2 above)
Comment (pzhanggit) [reply tests/test_graph_pred.py:45]: We want to check model loading and deployment as well as accuracy. The threshold is set to be equal to the maximum sample error we had in https://github.com/ORNL/HydraGNN/blob/215b5e8c93ff945e7da8951adc7fe9f5e3676c9c/tests/test_graphs.py#L126, since we are checking a single randomly selected sample. We could add one more to check the entire test set.

File: tests/test_graph_pred.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+##############################################################################
+import os, json
+import torch
+import random
+import hydragnn
+
+
+def unittest_model_pred(model_dir):
+    _, _ = hydragnn.utils.get_comm_size_and_rank()
+    # Read in config settings and override model type.
```
Comment (streeve) [inline tests/test_graph_pred.py:19]: Not overriding?

File: tests/test_graph_pred.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+#                                                                            #
+# SPDX-License-Identifier: BSD-3-Clause                                      #
+##############################################################################
+import os, json
+import torch
+import random
+import hydragnn
+
+
+def unittest_model_pred(model_dir):
```
Comment (streeve) [inline tests/test_graph_pred.py:17]: `unittest_model_prediction` + comment? This is just for testing prediction without training?

File: tests/test_graph_pred.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+    (
+        train_loader,
+        val_loader,
+        test_loader,
+        sampler_list,
+    ) = hydragnn.preprocess.load_data.dataset_loading_and_splitting(config=config)
+
+    model = hydragnn.models.create.create_model_config(
+        config=config["NeuralNetwork"]["Architecture"],
```
Comment (streeve) [inline tests/test_graph_pred.py:32]: This will have to change based on #127

File: tests/test_graphs.py [hunk 6]
```diff
@@ -16,6 +16,7 @@
 import shutil
 
 import hydragnn, tests
+from .test_graph_pred import unittest_model_pred
```
Comment (streeve) [inline tests/test_graphs.py:19]: Let's just do all of this in your new file

File: tests/test_graph_pred.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    config_file = os.path.join(model_dir, "config.json")
+    with open(config_file, "r") as f:
+        config = json.load(f)
+
+    (
+        train_loader,
+        val_loader,
+        test_loader,
+        sampler_list,
+    ) = hydragnn.preprocess.load_data.dataset_loading_and_splitting(config=config)
```
Comment (streeve) [inline tests/test_graph_pred.py:29]: Does this rely on another test having created the data? Or am I missing something

**Review by streeve — CHANGES_REQUESTED** [review]

File: tests/test_graph_pred.py (hunk 4 above)
Comment (pzhanggit) [reply tests/test_graph_pred.py:17]: Right.

File: tests/test_graph_pred.py (hunk 5 above)
Comment (pzhanggit) [reply tests/test_graph_pred.py:32]: Why? I do not see any changes needed after merging https://github.com/ORNL/HydraGNN/pull/127.

File: tests/test_graph_pred.py (hunk 7 above)
Comment (pzhanggit) [reply tests/test_graph_pred.py:29]: Yes, this test is for model loading and prediction check only.

File: tests/test_graphs.py (hunk 6 above)
Comment (pzhanggit) [reply tests/test_graphs.py:19]: Sure. One thing is that we want this test to run after the model training done in https://github.com/ORNL/HydraGNN/blob/bf1915c9ac3db508067f365289b674cf201bb7f5/tests/test_graphs.py#L183. Is a place where I can specify the order of unit tests?

File: tests/test_graph_pred.py (hunk 3 above)
Comment (pzhanggit) [reply tests/test_graph_pred.py:19]: Could you explain more?

File: tests/test_graphs.py (hunk 6 above)
Comment (streeve) [reply tests/test_graphs.py:19]: Let's use https://pytest-dependency.readthedocs.io/en/latest/usage.html#basic-usage :)

File: tests/test_graph_pred.py (hunk 3 above)
Comment (pzhanggit) [reply tests/test_graph_pred.py]: Removed.

File: tests/test_graphs.py (hunk 6 above)
Comment (pzhanggit) [reply tests/test_graphs.py]: It did not work as we expected, added a manual checking.

**Review by streeve — APPROVED** [review]
Looks good. Maybe we can revisit the pytest-dependency later on
